OutOfSchool/OutOfSchool.BusinessLogic/Services/CompetitiveEventService.cs

```diff
@@ -121,6 +121,55 @@ public async Task Delete(Guid id)
         }
     }
 
+    /// <inheritdoc/>
+    public async Task<SearchResult<CompetitiveEventViewCardDto>> GetByProviderId(Guid id, ExcludeIdFilter filter)
+    {
+        logger.LogDebug("Getting Competitive events by organization started. Looking ProviderId = {id}.", id);
+
```

Comment:
Use `ICurrentUserService.UserHasRights` method to check if user can actually get info about this provider's competitive events as this is provider only method.